CRM/Core/Payment/PayPalProIPN.php

```diff
@@ -611,16 +644,6 @@ public function handlePaymentExpress() {
       throw new CRM_Core_Exception('Failure: Could not find contribution record for ' . (int) $contribution->id, NULL, ['context' => "Could not find contribution record: {$contribution->id} in IPN request: " . print_r($input, TRUE)]);
     }
 
-    if (!empty($ids['contributionRecur'])) {
```

Comment:
this object is actually not used... it is a sort of validation that has been copied & pasted - but the getsingle above already does this validation